config/aso/kustomization.yaml

```diff
@@ -22,6 +22,15 @@ patches:
       version: v1
       kind: Deployment
       name: azureserviceoperator-controller-manager
+  - patch: |- # Add label for Azure workload identity webhook
+      - op: add
+        path: "/spec/template/metadata/labels/azure.workload.identity~1use"
```

Comment:
@mattchr Is this not needed per your point here or was that a separate issue? https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3113#discussion_r1123632661

Reply:
Sorry, wrong Matt. @matthchr

Reply:
> there's no reason to keep it from being added to ASO for consistency reasons.

This is true, but at the same time there is no reason to add it because it won't do anything for ASO. ASO doesn't need or use the WI webhook because it creates the credential without going to IMDS, which means it doesn't use the WI webhook at all.

Instead it uses the [USE_WORKLOAD_IDENTITY_AUTH](https://azure.github.io/azure-service-operator/guide/authentication/credential-format/) boolean for the global secret, or `AUTH_MODE:          "workloadidentity"` for the `aso-credential` or per-resource secrets.

The reason we don't use the webhook is that the webhook is based on env variables in the pod, but because the pod is multitenant (can be using multiple identities 1 per namespace or 1 per resource even) the webhook doesn't really help us.

Reply:
> Instead it uses the [USE_WORKLOAD_IDENTITY_AUTH](https://azure.github.io/azure-service-operator/guide/authentication/credential-format/) boolean for the global secret, or `AUTH_MODE: "workloadidentity"` for the `aso-credential` or per-resource secrets.

And CAPZ should configure this automatically for you based on the AzureClusterIdentity for the cluster.

Reply:
Thanks for all this context ! It makes sense. I can close this PR as it's not needed.